activejob/lib/active_job/serializers.rb

```diff
@@ -19,16 +19,17 @@ module Serializers # :nodoc:
     autoload :ModuleSerializer
     autoload :RangeSerializer
     autoload :BigDecimalSerializer
+    autoload :ActionControllerParametersSerializer
 
-    mattr_accessor :_additional_serializers
-    self._additional_serializers = Set.new
+    @serializers = Set.new
+    @serializers_index = {}
 
     class << self
       # Returns serialized representative of the passed object.
       # Will look up through all known serializers.
       # Raises ActiveJob::SerializationError if it can't find a proper serializer.
       def serialize(argument)
-        serializer = serializers.detect { |s| s.serialize?(argument) }
+        serializer = @serializers_index[argument.class] || serializers.find { |s| s.serialize?(argument) }
```

Comment:
I thought about adding an optimization like this but I was concerned it could interfere with a serializer who's `serialize?` method returns `true` or `false` based on the properties of the given object, for example:

```ruby
class CustomClass
  attr_reader :foo

  def initialize(foo)
    @foo = foo
  end
end

class CustomClassSerializer < ActiveJob::Serializers::ObjectSerializer
  def serialize?(argument)
    !!argument.foo
  end
end
```

Memoizing the mapping from class -> serializer would use `CustomClassSerializer` for instances of `CustomClass` that have a nil `@foo` and ones that don't.

I doubt this is a common use-case, but it is a change in behavior that could be subtle and difficult to debug for Rails devs.